semgrep-core/src/engine/Selector.ml

```diff
@@ -0,0 +1,130 @@
+module R = Rule
```

Comment:
Should we have an `.mli` for this module?

Reply:
Not a super fan of the name, Selector.ml.
Also, at some point we want to generalize this to handle different special cases, not just
pattern:$PROP but also pattern: include("foo") ..., and other things we think we can
specialize.